### ISSUE TITLE
Interface weirdness in _tab_regex

### ISSUE THREAD

**Issue by ZryletTC** [pcdshub/pcdsdevices#383]
## Expected Behavior
I expected _tab_regex to return the whitelisted component names.

## Current Behavior
From my tests, it appears that _tab_regex includes the values I have added with tab_whitelist (insert and remove in the below case):
`In [1]: xpp_sb2_ipm.diode.__class__._tab_regex                                                                                                                                                              
Out[1]: 
re.compile(r'insert|remove|name|connected|check_value|log|trigger|read|describe|stage|unstage|read_attrs|configuration_attrs|summary|wait_for_connection|stop|get|configure|value|put|settle_time|timeout|egu|limits|move|position|moving',
re.UNICODE)`
and when I comment out my tab_whitelist line, those values are gone as expected but every other value seems to have duplicated for some reason:
`In [1]: xpp_sb2_ipm.diode.__class__._tab_regex                                                                                                                                                             
Out[1]: 
re.compile(r'name|connected|check_value|log|trigger|read|describe|stage|unstage|read_attrs|configuration_attrs|summary|wait_for_connection|stop|get|configure|value|put|settle_time|timeout|egu|limits|move|position|moving|name|connected|check_value|log|trigger|read|describe|stage|unstage|read_attrs|configuration_attrs|summary|wait_for_connection|stop|get|configure|value|put|settle_time|timeout|egu|limits|move|position|moving',
re.UNICODE)`

Furthermore, changing the value of `tab_component_names` between `True` and `False` does not seem to affect the list. Either way, the above lists were shown and did not include the component names, whose `kind` was set to `normal`

## Possible Solution

🤷‍♂

## Steps to Reproduce (for bugs)

1. Run hutch-python
2. Look at an object's __class__._tab_regex
3. Add/remove tab_whitelist
4. Repeat steps 1&2 and compare output
5. Repeat steps 1-4 with the opposite value of tab_component_names

## Context
I'm creating some Ophyd classes and was trying to figure out how to set up my whitelist values when I stumbled across this.

## Your Environment
I tested this by changing values in the IPMDiode class in ipm.py in the dev pcdsdevices folder of and running xpppython from /reg/g/pcds/pyps/apps/hutch-python/xpp

**Comment by klauer** [comment]
Fix for the duplicate entries: https://github.com/pcdshub/pcdsdevices/blob/c7c4265dfcbf213b62a9a32039d2cf547edbfc1b/pcdsdevices/interface.py#L59 - change this to a `set` - though this shouldn't really affect anything other than the pattern's readability.

Side note: we may want to verify that the names passed in the tab whitelist are actually valid attribute names.

Local testing seems OK to me, but I might be missing your point:
```python
In [6]: class Test(pcdsdevices.interface.BaseInterface, ophyd.Device):
   ...:     tab_component_names = True
   ...:     cpt1 = ophyd.Component(ophyd.EpicsSignal, '')
   ...:

In [9]: 'cpt1' in Test.component_names
Out[9]: True

In [10]: 'cpt1' in Test._tab_regex.pattern
Out[10]: True

In [11]: class Test(pcdsdevices.interface.BaseInterface, ophyd.Device):
    ...:     tab_component_names = False
    ...:     cpt1 = ophyd.Component(ophyd.EpicsSignal, '')
    ...:

In [12]: 'cpt1' in Test.component_names
Out[12]: True

In [13]: 'cpt1' in Test._tab_regex.pattern
Out[13]: False

In [14]: Test._tab_regex.pattern
Out[14]: 'name|connected|check_value|log|trigger|read|describe|stage|unstage|read_attrs|configuration_attrs|summary|wait_for_connection|stop|get|configure|value|put|settle_time|timeout|egu|limits|move|position|moving|name|connected|check_value|log|trigger|read|describe|stage|unstage|read_attrs|configuration_attrs|summary|wait_for_connection|stop|get|configure|value|put|settle_time|timeout|egu|limits|move|position|moving'

In [19]: pcdsdevices.interface.set_engineering_mode(False)

In [20]: dir(a)
Out[20]:
['check_value', 'configuration_attrs', 'configure', 'connected', 'describe', 'get', 'log', 'name', 'put', 'read', 'read_attrs', 'stage', 'stop', 'summary', 'trigger', 'unstage', 'wait_for_connection']

```

**Comment by ZLLentz** [comment]
Maybe the issue is localized to certain classes somehow?

**Comment by klauer** [comment]
Should be fully resolved in #579
